docs/how-to-make-contribution-to-codebase.md

```diff
@@ -0,0 +1,216 @@
+Follow these guidelines to contribute to the codebase. This is highly recommended if you want to contribute regularly.
+
+Ignoring these steps will soil your copy which makes the contributing, maintaining, and reviewing processes difficult.
+
+## Making Contributing To Codebase
+
+You can now make changes to files and commit your changes to your fork, which you can prepare by reading [how to set up freecodecamp](how-to-setup-freecodecamp-locally.md).
+
+Follow these steps:
+
+1. Validate that you are on the `main` branch:
+
+   ```console
+   git status
+   ```
+
+   You should get an output like this:
+
+   ```console
+   On branch main
+   Your branch is up-to-date with 'origin/main'.
+
+   nothing to commit, working directory clean
+   ```
+
+   If you got different message, then you aren't on main or your working directory isn't clean, resolve any outstanding files/commits and checkout `main`:
+
+   ```console
+   git checkout main
+   ```
+
+2. Sync the latest changes from the freeCodeCamp upstream `main` branch to your `main` fork branch:
+
+   > [!WARNING]
+   > If you have any outstanding pull requests that you made from the `main` branch of your fork, you will lose them at the end of this step.
+   >
+   > You should ensure your pull request is merged by a moderator before performing this step. To avoid this scenario, you should **always** work on a branch other than the `main`.
+
+   This step **will sync the latest changes** from the main repository of freeCodeCamp. It is important that you rebase your branch on top of the latest `upstream/main` as often as possible to avoid conflicts later.
```

Comment:
"It is important that you rebase your branch on top of the latest `upstream/main` as often as possible to avoid conflicts later."

I didn't make a suggestion for this one. It feels like this line should be further down somewhere, but I didn't have a good place to suggest to add it.

Reply:
with some editing, this place seems fitting 🤔